pytorch_lightning/trainer/data_loading.py

```diff
@@ -459,3 +458,16 @@ def replace_sampler(dataloader: DataLoader) -> DataLoader:
             dataloader = apply_to_collection(dataloader, DataLoader, replace_sampler)
 
         return dataloader
+
+    @staticmethod
+    def _check_eval_shuffling(dataloader, mode):
+        if (
+            hasattr(dataloader, "sampler")
+            and not isinstance(dataloader.sampler, SequentialSampler)
+            and not isinstance(dataloader.dataset, IterableDataset)
+        ):
+            rank_zero_warn(
+                f"Your `{mode.dataloader_prefix}_dataloader` has `shuffle=True`,"
+                " it is strongly recommended that you turn this off for val/test/predict dataloaders.",
```

Comment:
Side question, do you remember why? Is that because `drop_last` could be set True? I usually toggle this as a sanity check to verify metrics are the same (shuffling should have no effect).

Reply:
yes, this is one case. another one can be if the user limits the val batches using `limit_val_batches`, so it's recommended to set `shuffle=False` all the time to be on the safe side.